src/app/clusters/testing/TestReadWriteAttribute.h

```diff
@@ -41,6 +42,16 @@ CHIP_ERROR ReadClusterAttribute(ClusterT & cluster, const app::ConcreteDataAttri
     return app::DataModel::Decode(attributeData[0].dataReader, value);
 }
 
+// Helper function to read any attribute value of a given type
+template <typename ClusterT, typename T>
+CHIP_ERROR ReadClusterAttribute(ClusterT & cluster, AttributeId attr, T & val)
```

Comment:
```suggestion
template <typename T>
CHIP_ERROR ReadClusterAttribute(DefaultServerCluster & cluster, AttributeId attr, T & val)
```

Reasoning being that a default server cluster is KNOWN to have exactly 1 path and then the whole thing below is generally guaranteed ok. We could also make it `ServerClusterInterface` however in practice we do not really have clusters that are not defaultservercluster.

Overall I wonder why we have a template above in the first place...

Also I would say that the above function is NOT safe... can we please fix that? The reason being:

- we read data into `std::vector` of decodeded data.
- Reads may reference into this data (strings and lists will do so)
- returned `value` will the referenced freed memory for the stack.

Please add a static_assert that the value is integral or enum